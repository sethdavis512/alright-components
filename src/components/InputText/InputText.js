import React from 'react';
import PropTypes from 'prop-types';

function InputText({ label, onChange, value }) {
    return (
        <label>
            {label}
            <input className="input" onChange={onChange} value={value} />
        </label>
    );
}

InputText.propTypes = {
    label: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    value: PropTypes.string.isRequired
};

InputText.defaultProps = {};

export default InputText;
