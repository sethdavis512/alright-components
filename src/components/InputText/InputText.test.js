import React from 'react';
import { shallow } from 'enzyme';
import InputText from './InputText';

describe('InputText', () => {
    const defaultProps = {
        label: 'Default Label',
        onChange: jest.fn(),
        value: ''
    };

    function renderComponent(props) {
        return shallow(<InputText {...defaultProps} {...props} />);
    }

    it('should match snapshot', () => {
        const wrapper = renderComponent();
        expect(wrapper).toMatchSnapshot();
    });
});
