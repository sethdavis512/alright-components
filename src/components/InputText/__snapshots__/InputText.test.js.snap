// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText should match snapshot 1`] = `
<label>
  Default Label
  <input
    className="input"
    onChange={[MockFunction]}
    value=""
  />
</label>
`;
